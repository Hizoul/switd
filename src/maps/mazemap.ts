import Map, { experimentChoices } from "../logic/map"
import Street, { direction } from "../logic/street"
import Tower from "../logic/tower"

const createtestMapNoSplits = () => {
  const testMapNoSplits = new Map(15, 15)
  testMapNoSplits.spawnThreshold = 14
  testMapNoSplits.experimentType = experimentChoices.shortestPathOnly
  const start = new Street(0, 0, testMapNoSplits)
  testMapNoSplits.addTile(start)
  const s1 = start.extend(direction.right)
  const s2 = s1.extend(direction.right)
  const s3 = s2.extend(direction.right)
  const s4 = s3.extend(direction.right)
  const s5 = s4.extend(direction.right)
  const s6 = s5.extend(direction.right)
  const s7 = s6.extend(direction.right)
  const s8 = s7.extend(direction.right)
  const s9 = s8.extend(direction.right)
  const s10 = s9.extend(direction.right)
  const s11 = s10.extend(direction.right)
  const s12 = s11.multiExtend(direction.down, 4)
  const s13 = s12.multiExtend(direction.left, 1)

  const s2a = s2.extend(direction.down)
  const s2b = s2a.extend(direction.down)
  const s2ca = s2b.multiExtend(direction.right, 3)
  const s2cb = s2ca.multiExtend(direction.right, 4)
  const s2d = s2cb.multiExtend(direction.down, 2)
  const s2e = s2d.multiExtend(direction.down, 5)

  s13.addIntersection(s2d)

  const w3 = s2ca.extend(direction.down)
  const w4 = w3.extend(direction.down)
  const w5 = w4.multiExtend(direction.right, 3)

  w5.addIntersection(s2d)

  const t1 = new Tower(4, 6, testMapNoSplits)
  t1.damagableAnts = 4
  t1.damageAmount = 60
  t1.towerRange = 2
  testMapNoSplits.addTile(t1)

  return testMapNoSplits

}

export default createtestMapNoSplits()
export { createtestMapNoSplits }
