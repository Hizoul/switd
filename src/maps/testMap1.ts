import Map from "../logic/map"
import Street, { direction } from "../logic/street"

const createtestMap1 = () => {
  const testMap1 = new Map()
  const start = new Street(0, 0, testMap1)
  testMap1.addTile(start)
  const s1 = start.extend(direction.right)
  const s2 = s1.extend(direction.down)

  const s31 = s2.extend(direction.down)
  const s311 = s31.extend(direction.down)
  const s312 = s311.extend(direction.down)
  const s313 = s312.extend(direction.right)
  const s314intersection = s313.extend(direction.right)

  const s32 = s2.extend(direction.right)
  const s321 = s32.extend(direction.right)
  const s322 = s321.extend(direction.down)
  const s323 = s322.extend(direction.down)
  s323.addIntersection(s314intersection)
  const s4 = s314intersection.extend(direction.down)
  const s5 = s4.extend(direction.down)
  const s6 = s5.extend(direction.right)
  return testMap1
}

export default createtestMap1()
export { createtestMap1 }
