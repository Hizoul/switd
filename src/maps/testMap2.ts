import Map from "../logic/map"
import Street, { direction } from "../logic/street"
import Tower from "../logic/tower"

const createtestMap2 = () => {
  const testMap2 = new Map()
  testMap2.spawnThreshold = 14
  const start = new Street(0, 0, testMap2)
  testMap2.addTile(start)
  const s1 = start.extend(direction.right)
  const s2 = s1.extend(direction.down)

  const s31 = s2.extend(direction.down)
  const s311 = s31.extend(direction.down)
  const s312 = s311.extend(direction.down)
  const s313 = s312.extend(direction.right)
  const s314 = s313.extend(direction.right)
  const s315intersection = s314.extend(direction.right)

  const s32 = s2.extend(direction.right)
  const s321 = s32.extend(direction.right)
  const s322 = s321.extend(direction.right)
  const s323 = s322.extend(direction.down)
  const s324 = s323.extend(direction.down)
  s324.addIntersection(s315intersection)
  const s4 = s315intersection.extend(direction.down)
  const s5 = s4.extend(direction.down)
  const s6 = s5.extend(direction.right)

  const t1 = new Tower(2, 2, testMap2)
  testMap2.addTile(t1)
  const t2 = new Tower(3, 3, testMap2)
  testMap2.addTile(t2)
  const t3 = new Tower(5, 5, testMap2)
  testMap2.addTile(t3)

  return testMap2
}

export default createtestMap2()
export { createtestMap2 }
