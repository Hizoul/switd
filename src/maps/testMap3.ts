import Map from "../logic/map"
import Street, { direction } from "../logic/street"
import Tower from "../logic/tower"

const createtestMap3 = () => {
  const testMap2 = new Map()
  testMap2.spawnThreshold = 14
  const start = new Street(0, 0, testMap2)
  testMap2.addTile(start)
  const s1 = start.extend(direction.right)
  const s2 = s1.extend(direction.down)

  const s31 = s2.extend(direction.down)
  const s311 = s31.extend(direction.down)
  const s312 = s311.extend(direction.down)
  const s313 = s312.extend(direction.right)
  const s314 = s313.extend(direction.right)
  const s315intersection = s314.extend(direction.right)

  const s32 = s2.extend(direction.right)

  testMap2.intersectionList.push({
    beginning: s31, target: s32
  })

  const s321 = s32.extend(direction.right)
  const s322 = s321.extend(direction.right)
  const s323 = s322.extend(direction.down)
  const s324 = s323.extend(direction.down)
  s324.addIntersection(s315intersection)
  const s4 = s315intersection.extend(direction.down)
  const s5 = s4.extend(direction.down)
  const s6 = s5.extend(direction.right)

  const t2 = new Tower(2, 3, testMap2)
  testMap2.addTile(t2)
  const t3 = new Tower(5, 5, testMap2)
  testMap2.addTile(t3)

  return testMap2
}

export default createtestMap3()
export { createtestMap3 }
