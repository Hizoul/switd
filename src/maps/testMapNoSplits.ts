import Map from "../logic/map"
import Street, { direction } from "../logic/street"
import Tower from "../logic/tower"

const createtestMapNoSplits = () => {
  const testMapNoSplits = new Map()
  testMapNoSplits.spawnThreshold = 14
  const start = new Street(0, 0, testMapNoSplits)
  testMapNoSplits.addTile(start)
  const s1 = start.extend(direction.right)
  const s2 = s1.extend(direction.down)
  const s3 = s2.extend(direction.down)
  const s4 = s3.extend(direction.right)
  const s5 = s4.extend(direction.right)
  const s6 = s5.extend(direction.right)
  const s7 = s6.extend(direction.down)
  const s8 = s7.extend(direction.down)
  const s9 = s8.extend(direction.left)
  const s10 = s9.extend(direction.left)
  const s11 = s10.extend(direction.left)
  const t1 = new Tower(2, 1, testMapNoSplits)
  testMapNoSplits.addTile(t1)
  const t2 = new Tower(3, 3, testMapNoSplits)
  testMapNoSplits.addTile(t2)
  const t3 = new Tower(4, 5, testMapNoSplits)
  testMapNoSplits.addTile(t3)
  return testMapNoSplits
}

export default createtestMapNoSplits()
export { createtestMapNoSplits }
