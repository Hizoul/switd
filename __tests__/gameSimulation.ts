import { createtestMap1 } from "../src/maps/testMap1"
import prepSnapshotOutput from "../src/testUtil/mapSnapshot"

it("A gamefield can generate ants and let them walk in a tick", () => {
  const testedMap = createtestMap1()
  expect(prepSnapshotOutput(testedMap)).toMatchSnapshot("no ants yet")
  testedMap.spawnNewAnt(5)
  expect(prepSnapshotOutput(testedMap)).toMatchSnapshot("five unmoved ants")
  testedMap.processTick()
  testedMap.spawnNewAnt(5)
  expect(prepSnapshotOutput(testedMap)).toMatchSnapshot("five ants at stage 1 5 more newly added")
  testedMap.processTick()
  expect(prepSnapshotOutput(testedMap)).toMatchSnapshot("five ants at stage 2 5 more at 1")
})
