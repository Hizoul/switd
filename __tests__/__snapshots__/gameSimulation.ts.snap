// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`five ants at stage 1 5 more newly added 1`] = `
Object {
  "ants": Array [
    Object {
      "id": 1,
      "x": 1,
      "y": 0,
    },
    Object {
      "id": 2,
      "x": 1,
      "y": 0,
    },
    Object {
      "id": 3,
      "x": 1,
      "y": 0,
    },
    Object {
      "id": 4,
      "x": 1,
      "y": 0,
    },
    Object {
      "id": 5,
      "x": 1,
      "y": 0,
    },
    Object {
      "id": 6,
      "x": 0,
      "y": 0,
    },
    Object {
      "id": 7,
      "x": 0,
      "y": 0,
    },
    Object {
      "id": 8,
      "x": 0,
      "y": 0,
    },
    Object {
      "id": 9,
      "x": 0,
      "y": 0,
    },
    Object {
      "id": 10,
      "x": 0,
      "y": 0,
    },
  ],
  "tiles": Array [
    Object {
      "x": 0,
      "y": 0,
    },
    Object {
      "x": 1,
      "y": 0,
    },
    Object {
      "x": 1,
      "y": -1,
    },
    Object {
      "x": 1,
      "y": -2,
    },
    Object {
      "x": 1,
      "y": -3,
    },
    Object {
      "x": 1,
      "y": -4,
    },
    Object {
      "x": 2,
      "y": -4,
    },
    Object {
      "x": 3,
      "y": -4,
    },
    Object {
      "x": 2,
      "y": -1,
    },
    Object {
      "x": 3,
      "y": -1,
    },
    Object {
      "x": 3,
      "y": -2,
    },
    Object {
      "x": 3,
      "y": -3,
    },
    Object {
      "x": 3,
      "y": -5,
    },
    Object {
      "x": 3,
      "y": -6,
    },
    Object {
      "x": 4,
      "y": -6,
    },
  ],
}
`;

exports[`five ants at stage 2 5 more at 1 1`] = `
Object {
  "ants": Array [
    Object {
      "id": 1,
      "x": 1,
      "y": -1,
    },
    Object {
      "id": 2,
      "x": 1,
      "y": -1,
    },
    Object {
      "id": 3,
      "x": 1,
      "y": -1,
    },
    Object {
      "id": 4,
      "x": 1,
      "y": -1,
    },
    Object {
      "id": 5,
      "x": 1,
      "y": -1,
    },
    Object {
      "id": 6,
      "x": 1,
      "y": 0,
    },
    Object {
      "id": 7,
      "x": 1,
      "y": 0,
    },
    Object {
      "id": 8,
      "x": 1,
      "y": 0,
    },
    Object {
      "id": 9,
      "x": 1,
      "y": 0,
    },
    Object {
      "id": 10,
      "x": 1,
      "y": 0,
    },
  ],
  "tiles": Array [
    Object {
      "x": 0,
      "y": 0,
    },
    Object {
      "x": 1,
      "y": 0,
    },
    Object {
      "x": 1,
      "y": -1,
    },
    Object {
      "x": 1,
      "y": -2,
    },
    Object {
      "x": 1,
      "y": -3,
    },
    Object {
      "x": 1,
      "y": -4,
    },
    Object {
      "x": 2,
      "y": -4,
    },
    Object {
      "x": 3,
      "y": -4,
    },
    Object {
      "x": 2,
      "y": -1,
    },
    Object {
      "x": 3,
      "y": -1,
    },
    Object {
      "x": 3,
      "y": -2,
    },
    Object {
      "x": 3,
      "y": -3,
    },
    Object {
      "x": 3,
      "y": -5,
    },
    Object {
      "x": 3,
      "y": -6,
    },
    Object {
      "x": 4,
      "y": -6,
    },
  ],
}
`;

exports[`five unmoved ants 1`] = `
Object {
  "ants": Array [
    Object {
      "id": 1,
      "x": 0,
      "y": 0,
    },
    Object {
      "id": 2,
      "x": 0,
      "y": 0,
    },
    Object {
      "id": 3,
      "x": 0,
      "y": 0,
    },
    Object {
      "id": 4,
      "x": 0,
      "y": 0,
    },
    Object {
      "id": 5,
      "x": 0,
      "y": 0,
    },
  ],
  "tiles": Array [
    Object {
      "x": 0,
      "y": 0,
    },
    Object {
      "x": 1,
      "y": 0,
    },
    Object {
      "x": 1,
      "y": -1,
    },
    Object {
      "x": 1,
      "y": -2,
    },
    Object {
      "x": 1,
      "y": -3,
    },
    Object {
      "x": 1,
      "y": -4,
    },
    Object {
      "x": 2,
      "y": -4,
    },
    Object {
      "x": 3,
      "y": -4,
    },
    Object {
      "x": 2,
      "y": -1,
    },
    Object {
      "x": 3,
      "y": -1,
    },
    Object {
      "x": 3,
      "y": -2,
    },
    Object {
      "x": 3,
      "y": -3,
    },
    Object {
      "x": 3,
      "y": -5,
    },
    Object {
      "x": 3,
      "y": -6,
    },
    Object {
      "x": 4,
      "y": -6,
    },
  ],
}
`;

exports[`no ants yet 1`] = `
Object {
  "ants": Array [],
  "tiles": Array [
    Object {
      "x": 0,
      "y": 0,
    },
    Object {
      "x": 1,
      "y": 0,
    },
    Object {
      "x": 1,
      "y": -1,
    },
    Object {
      "x": 1,
      "y": -2,
    },
    Object {
      "x": 1,
      "y": -3,
    },
    Object {
      "x": 1,
      "y": -4,
    },
    Object {
      "x": 2,
      "y": -4,
    },
    Object {
      "x": 3,
      "y": -4,
    },
    Object {
      "x": 2,
      "y": -1,
    },
    Object {
      "x": 3,
      "y": -1,
    },
    Object {
      "x": 3,
      "y": -2,
    },
    Object {
      "x": 3,
      "y": -3,
    },
    Object {
      "x": 3,
      "y": -5,
    },
    Object {
      "x": 3,
      "y": -6,
    },
    Object {
      "x": 4,
      "y": -6,
    },
  ],
}
`;
